### PR TITLE
History page

### PR DESCRIPTION
**Pull request by agrawalshanu708**
History page ADDED

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -13,6 +12,7 @@ function App() {
       <Route path="/" element = {<Home/>}/>
```
Comment (Chiragmodi01) [inline src/App.js:12]: You can also add 404 page, it'll look good.

File: src/backend/db/videos.js [hunk 2]
```diff
@@ -5,19 +5,20 @@
 
 export const videos = [
 {
-  id: "vi/-yLlMk41sro",
+  _id: "m050iy5_2ng",
```
Comment (Chiragmodi01) [inline src/backend/db/videos.js:8]: Not sure why you've used static data here but prefer uuid.

File: src/backend/db/videos.js [hunk 3]
```diff
@@ -5,19 +5,20 @@
...
   title: "talks",
   charactor: "sunadar pichai",
   description: "",
```
Comment (Chiragmodi01) [inline src/backend/db/videos.js:11]: desc is empty!

File: src/components/addedvideo/addedvideo.jsx [hunk 4]
```diff
@@ -3,11 +3,11 @@ import "./addedvideo.css"
 const AddedVideo = () => {
   return (
     <>
-    <div className="addedvideo col-3">
+    <div className="addedvideo col-2">
         <p>New video</p>
         <div className="addedvideo__container">
           {
-            
+
```
Comment (Chiragmodi01) [inline src/components/addedvideo/addedvideo.jsx:10]: add logic here.

File: src/components/videostore/videocard.jsx [hunk 5]
```diff
@@ -1,41 +1,60 @@
...
-      />
+      <div class="text-overlay-card-img-box" onClick={() => historyDispatch({
+        type: "ADD_IN_HISTORY",
+        payload: product
+      })}>
+        <img
+          src={`https://i.ytimg.com/vi/${_id}/hqdefault.jpg`}
+        />
       </div>
+      {/* <div class="card-head head-overflow">{title}</div> */}
```
Comment (Chiragmodi01) [inline src/components/videostore/videocard.jsx:42]: remove commented code

File: src/components/videostore/videocard.jsx [hunk 6]
```diff
@@ -1,41 +1,60 @@
...
+      <div class="card-tag tag-overflow ">{charactor}</div>
+      <div class="card-desc skin">
+        Visit ten places on our planet that are undergoing the biggest
+        changes today</div>
+      <div class="card-footer-box card__icons">
+        <AiOutlineHeart color="#AB542F" size="3rem" onClick={() => likeDispatch({
+          type: "ADD_TO_LIKE",
+          payload: product
+        })} />
+        <MdOutlinePlaylistPlay color="#ffff" size="4rem" />
```
Comment (Chiragmodi01) [inline src/components/videostore/videocard.jsx:52]: I prefer styling color in css file, only if icon has few more style properties.

File: src/components/videostore/videocard.jsx [hunk 7]
```diff
@@ -1,41 +1,60 @@
...
+        <MdOutlineWatchLater color="#AB542F" size="3rem"  onClick={() =>  {
```
Comment (Chiragmodi01) [inline src/components/videostore/videocard.jsx:53]: Use variable for colors in css file, cause i don't know what this color is .

File: src/index.js [hunk 8]
```diff
@@ -4,7 +4,7 @@ import "./index.css";
 import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
-import { LikeProvider,WatchLaterProvider } from "./context/index";
+import { LikeProvider,WatchLaterProvider,HistoryProvider} from "./context/index";
 
 
 // Call make Server
```
Comment (Chiragmodi01) [inline src/index.js:10]: (comment without the logic)

File: src/pages/historypage/history.css [hunk 9]
```diff
@@ -0,0 +1,11 @@
+.historypage{
+    display: flex;
+    gap: 1rem;
+    flex-wrap: wrap;
+}
+.card_close{
+    position: absolute;
+    top:2%;
+    right:2%;
```
Comment (Chiragmodi01) [inline src/pages/historypage/history.css:9]: You can use `px` for small units. Use `%` for larger distance.

File: src/pages/historypage/historycard.jsx [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    const{_id,title,description,charactor} = product
+
+  return (
+<div class="border-skin text-overlay-card-dimension card-relative video-card" key = {index}>
+    <div class="text-overlay-card-img-box">
+      <img
+      src={`https://i.ytimg.com/vi/${_id}/hq720.jpg`} class="history_media"
+    />
+    </div>
+    {/* <div class="card-head head-overflow">{title}</div> */}
```
Comment (Chiragmodi01) [inline src/pages/historypage/historycard.jsx:16]: remove commented code before pushing, else you can maintain separate 'notes file' for commented or trash code.

File: src/pages/historypage/historycard.jsx [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    <AiOutlineClose color="#AB542F" size="4rem" class="card_close"
```
Comment (Chiragmodi01) [inline src/pages/historypage/historycard.jsx:17]: I prefer sizing icons using 'em' units, cause they are relative to parent's font size.

File: src/backend/db/videos.js [hunk 12]
```diff
@@ -5,19 +5,20 @@
...
   link: "https://www.youtube.com/watch?v=-yLlMk41sro"
 },{
-  id: "Lb4IcGF5iTQ",
+  _id: "Lb4IcGF5iTQ",
   title: "talks",
   charactor: "Mark Zuckerberg",
   description: "",
```
Comment (Sharmaryan) [inline src/backend/db/videos.js:17]: description value is empty

File: src/pages/historypage/historycard.jsx [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    onClick = {() => historyDispatch({
+      type: "REMOVE_FROM_HISTORY",
+      payload: _id
+    })}/>
+    <div class="card-tag tag-overflow ">{charactor}</div>
```
Comment (Chiragmodi01) [inline src/pages/historypage/historycard.jsx:22]: can use 'char' than `character`.

File: src/pages/home/home.jsx [hunk 14]
```diff
@@ -4,7 +4,7 @@ const Home = () => {
   return (
 
     <>
-    <div className="home">
+    <div className="homepage col-11">
```
Comment (Chiragmodi01) [inline src/pages/home/home.jsx:7]: great that you're using CL's classes for spacing in here!

**Review by Sharmaryan — APPROVED** [review]
- UI suggestion : you can add cursor: pointer on video card, like button, watchlater
- UI looks  good

suggested small changes

**Review by Chiragmodi01 — COMMENTED** [review]
Overall code structure looks good, just added few sanity comments.
In UI the card looks a bit clustered to me, you can polish it with some spacing here and there. in the app. The whole app is currently looking static'ish in look to me.

**Pull request merged**